tests/std/tests/P1147R1_printing_volotile_pointers/test.cpp

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <cstddef>
+#include <sstream>
+
+using namespace std;
+
+std::string getExpected(int* ptr) {
+    ostringstream out;
+    out << ptr;
+    return out.str();
+}
+
+std::string getActual(volatile int* ptr) {
+    ostringstream out;
+    out << ptr;
+    return out.str();
+}
+
+void test(size_t value) {
+    int* p0          = reinterpret_cast<int*>(value);
+    volatile int* p1 = reinterpret_cast<volatile int*>(p0);

```

Comment:
pointer-to-`int` converts implicitly to pointer-to-`volatile int`; the cast here is unnecessary.